DiscordChatExporter.Domain/Discord/Models/MessageReference.cs

```diff
@@ -0,0 +1,37 @@
+using System.Text.Json;
+using JsonExtensions.Reading;
+
+namespace DiscordChatExporter.Domain.Discord.Models
+{
+    // reference data sent with crossposted messages and replies
+    // https://discord.com/developers/docs/resources/channel#message-object-message-reference-structure
+    public partial class MessageReference
+    {
+        public string? MessageId { get; }
+
+        public string? ChannelId { get; }
+
+        public string? GuildId { get; }
```

Comment:
Discord API docs claim that all of these can be nullable, but it's a bit weird to me. I think we should be able to assume that these are actually not nullable.